src/audio/copier/copier.c

```diff
@@ -165,26 +127,16 @@ static struct comp_dev *copier_new(const struct comp_driver *drv,
 				comp_cl_err(&comp_copier, "unable to create dai");
 				goto error_cd;
 			}
-
-			if (cd->direction == SOF_IPC_STREAM_PLAYBACK)
-				ipc_pipe->pipeline->sink_comp = dev;
-			else
-				ipc_pipe->pipeline->source_comp = dev;
-
 			break;
 #if CONFIG_IPC4_GATEWAY
 		case ipc4_ipc_output_class:
 		case ipc4_ipc_input_class:
-			if (copier_ipcgtw_create(dev, cd, &dev->ipc_config, copier)) {
+			if (copier_ipcgtw_create(dev, cd, &dev->ipc_config,
+						 copier, ipc_pipe->pipeline)) {
 				comp_cl_err(&comp_copier, "unable to create IPC gateway");
 				goto error_cd;
 			}
 
-			if (cd->direction == SOF_IPC_STREAM_PLAYBACK)
-				ipc_pipe->pipeline->source_comp = dev;
-			else
-				ipc_pipe->pipeline->sink_comp = dev;
```

Comment:
this doesn't need to be added to `copier_ipcgtw_create()`?

Reply:
any reason that this part can't move to copier_ipcgtw_create?
move to create function can make copier_new looks as simple as possible.
Also, inside can combine with other PLAYBACK and CAPTURE case.